sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -801,8 +818,8 @@ object ExpressionContext {
   }
 
   def getRegularOperatorContext(meta: RapidsMeta[_, _, _]): ExpressionContext = meta.wrapped match {
-    case _: LambdaFunction => LambdaExprContext
     case _: Expression if meta.parent.isDefined => getRegularOperatorContext(meta.parent.get)
+    // TODO do we want to look for joins/etc here and check the join type???
```

Comment:
I'm not sure I follow how the join type matters to the AST expression. It's certainly critical for the join to know whether the condition expression can be translated to an AST, but I'm not following why the expr meta needs to check for joins here and why expression behavior would change based on join type.

Reply:
Sorry this was left over from me trying to decide if we try to auto detect the `AstContext` or treat it totally separately. I went with totally separately.